api4/post_test.go

```diff
@@ -2527,3 +2527,84 @@ func TestGetFileInfosForPost(t *testing.T) {
 	_, resp = th.SystemAdminClient.GetFileInfosForPost(th.BasicPost.Id, "")
 	CheckNoError(t, resp)
 }
+
+func TestSetChannelUnread(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	u1 := th.BasicUser
+	u2 := th.BasicUser2
+	s2, _ := th.App.GetSession(th.Client.AuthToken)
+	th.Client.Login(u1.Email, u1.Password)
+	c1 := th.BasicChannel
+	c1toc2 := &model.ChannelView{ChannelId: th.BasicChannel2.Id, PrevChannelId: c1.Id}
+	now := utils.MillisFromTime(time.Now())
+	p1 := th.CreateMessagePostNoClient(c1, "AAA", now)
+	require.NotNil(t, p1)
+	p2 := th.CreateMessagePostNoClient(c1, "BBB", now+10)
+	require.NotNil(t, p2)
+	p3 := th.CreateMessagePostNoClient(c1, "CCC", now+20)
+	require.NotNil(t, p3)
+
+	pp1 := th.CreateMessagePostNoClient(th.BasicPrivateChannel, "Sssh!", now)
+	pp2 := th.CreateMessagePostNoClient(th.BasicPrivateChannel, "You Sssh!", now+10)
+	require.NotNil(t, pp1)
+	require.NotNil(t, pp2)
+
+	t.Run("Can't unread if user is not logged in", func(t *testing.T) {
+		th.Client.Logout()
+		response := th.Client.SetPostUnread(u1.Id, p2.Id)
+		checkHTTPStatus(t, response, http.StatusUnauthorized, true)
+	})
+
+	th.Client.Login(u1.Email, u1.Password)
+
+	t.Run("Ensure that post have been read", func(t *testing.T) {
```

Comment:
Maybe remove the `t.Run` here since we're just confirming that the initial conditions are correct for this set of tests? This isn't testing `SetPostUnread` itself, so I wouldn't expect to see it wrapped in `t.Run`.